text/0000-ques-in-main.md

```diff
@@ -0,0 +1,514 @@
+section talking about interoperation with the Unix command-line
+ecosystem.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Generalizing the return type of `main` complicates libstd and/or the
+compiler.  It also adds an additional thing to remember when complete
+newbies to the language get to error handling.  On the other hand,
+people coming to Rust from other languages may find this _less_
+surprising than the status quo.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Do nothing; continue to live with the trip hazard, the extra
+boilerplate required to comply with platform conventions, and people
+using `panic!` to report ordinary errors because it's less hassle.
+
+The [pre-RFC][pre-rfc] included a suggestion to use `catch` instead,
+but this still involves extra boilerplate in `main` so I'm not
+enthusiastic about it.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+We need to decide what to call the new trait.  The names proposed in
+the pre-RFC thread were `Terminate`, which I like OK but have changed
+to `Termination` because return value traits should be nouns, and
+`Fallible`, which feels much too general, but could be OK if there
+were other uses for it?  Relatedly, it is conceivable that there are
+other uses for `Termination` in the existing standard library, but I
+can't think of any right now.  (Thread join was mentioned in the
+[pre-RFC][pre-rfc], but that can already relay anything that's `Send`,
+so I don't see that it adds value there.)
+
+We also need to decide where the trait should live.  One obvious place
+is in `std::process`, because that is where `exit(i32)` is; on the
+other hand, this is basic enough that it may make sense to put at
+least some of it in libcore.
+
+I don't know what impls of `Termination` should be available beyond
+the ones listed above, nor do I know what impls should be in libcore.
+Most importantly, I do not know whether it is necessary to impl
+Termination for `Box<T> where T: Termination`.  It might be that Box's
+existing impl of Deref renders this unnecessary.
+
+It may make sense to provide a type alias like
+
+    type MayFail = Result<(), Box<Error>>;
+
+which would mean less boilerplate in the most common case for `main`.
+However, this may be too single-purpose for a global-prelude type
+alias, and if we ever get return type deduction, that would completely
+supersede it, but we'd have burned the name forever.
+
+There is an outstanding proposal to [generalize `?`][try-trait]
+(see also RFC issues [#1718][rfc-i1718] and [#1859][rfc-i1859]); I
+think it is mostly orthogonal to this proposal, but we should make
+sure it doesn't conflict and we should also figure out whether we
+would need more impls of `Termination` to make them play well
+together.
+
+Most operating systems accept only a scalar exit status, but
+[Plan 9][], uniquely (to my knowledge), takes an entire string (see
+[`exits(2)`][exits.2]).  Do we care?  If we care, what do we do about
+it?
```

Comment:
Plan 9 is not in https://forge.rust-lang.org/platform-support.html, I guess not (yet).

Plus, an integer can be easily formatted to a string.